source/PreferencesPanel.cpp

```diff
@@ -583,27 +586,49 @@ void PreferencesPanel::DrawPlugins()
 	const Color &back = *GameData::Colors().Get("faint");
 	const Color &medium = *GameData::Colors().Get("medium");
 	const Color &bright = *GameData::Colors().Get("bright");
+	const Color &error = *GameData::Colors().Get("bright error");
+
+	const Sprite *box[2] = { SpriteSet::Get("ui/unchecked"), SpriteSet::Get("ui/checked") };
 
 	const int MAX_TEXT_WIDTH = 230;
 	Table table;
 	table.AddColumn(-115, {MAX_TEXT_WIDTH, Truncate::MIDDLE});
-	table.SetUnderline(-120, 120);
+	table.SetUnderline(-120, 100);
 
 	int firstY = -238;
-	table.DrawAt(Point(-130, firstY));
+	// Table is at -110 while checkbox is at -130
+	table.DrawAt(Point(-110, firstY));
 	table.DrawUnderline(medium);
 	table.Draw("Installed plugins:", bright);
 	table.DrawGap(5);
 
 	const Font &font = FontSet::Get(14);
+
 	for(const auto &plugin : GameData::PluginAboutText())
 	{
 		pluginZones.emplace_back(table.GetCenterPoint(), table.GetRowSize(), plugin.first);
 
 		bool isSelected = (plugin.first == selectedPlugin);
 		if(isSelected || plugin.first == hoverPlugin)
 			table.DrawHighlight(back);
-		table.Draw(plugin.first, isSelected ? bright : medium);
+		bool pluginEnabled = Plugins::IsEnabled(plugin.first);
+		double rowHeight = table.GetRowBounds().Height();
+		double thirdHeight = rowHeight / 3;
+		Point checkboxPos = table.GetRowBounds().TopLeft();
+		checkboxPos.X() -= font.Height() / 2;
+		checkboxPos.Y() += rowHeight / 2;
+
+		SpriteShader::Draw(box[pluginEnabled], checkboxPos);
+		Point zoneDimension = Point(20., thirdHeight * 2.);
+		Point zoneOffset = checkboxPos + Point(0., thirdHeight - 3);
+		Rectangle zoneBounds = Rectangle(zoneOffset, zoneDimension);
```

Comment:
It might make sense to add a Rectangle constructor that takes an existing rectangle and then the amount to expand or contract its vertical and horizontal dimensions.

Reply:
This was trial and error guessing.  Both for code and dimensions.  I have no meaningful way to adjust the UI code other than changing values and building.

I will need a bit more hand holding on this request because I don’t understand and am new to graphics code.

Reply:
My comment was more of a "geometric operations simplification" request, since you're creating a box from another box, with a bunch of intermediate steps.

Reply:
I couldn't seem to get the boxes to line up initially and that's what all the extra math is for.  It was trial and error because one appears to use the center while the other uses some other dimension I couldn't figure out because it was invisible (maybe a corner?).

The current sizing was me trying to work within those constraints while not quite understanding why the two shapes didn't line up originally.

Reply:
If we do this I think it should be a separate PR.